omc3/plotting/utils/annotations.py

```diff
@@ -298,17 +305,31 @@ def reorder(items):
     return reorder(handles), reorder(labels)
 
 
-def make_top_legend(ax, ncol, frame=False, handles=None, labels=None, pad=0.02, transposed=False):
+def make_top_legend(
+    ax: Axes, ncol: int, frame: bool = False, handles=None, labels=None, pad: float = 0.02, transposed=False
+):
     """Create a legend on top of the plot."""
     if ncol < 1:
         return
 
     if transposed:
         handles, labels = transpose_legend_order(ncol, handles, labels, ax)
 
-    leg = ax.legend(handles=handles, labels=labels, loc='lower right',
-                    bbox_to_anchor=(1.0, 1.0+pad),
-                    fancybox=frame, shadow=frame, frameon=frame, ncol=ncol)
+    # It is possible that no labels are given to the legend commands, or they are
+    # given with a leading underscore, which will get them ignored. In this case
+    # matplotlib issues a UserWarning which we will suppress here
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", UserWarning)
+        leg = ax.legend(
+            handles=handles,
+            labels=labels,
+            loc="lower right",
+            bbox_to_anchor=(1.0, 1.0+pad),
+            fancybox=frame,
+            shadow=frame,
+            frameon=frame,
+            ncol=ncol,
+        )
```

Comment:
**Note**: Do we want to keep these showing up?

This relates to the `UserWarning: No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.`.

I assumed most of the time this happens because there is a leading underscore, or we didn't write a label for every / any lines. We might want to be more thorough and write those every time instead of filtering the warning here.

Reply:
I think this is a reasonable warning, and should be taken care of in the test that raises it. I.e. assign some labels or do not create the legend.

Reply:
I reverted the intercept. Did not manage to figure out where in the tests we were encountering this specific condition, so for now it will stay in the outputs.